### PR TITLE
Upgrade to xenial

### PR DESCRIPTION
**Pull request by psivesely**
Upgrading our Vagrant image to Xenial would mean that (i) we wouldn't have to rely on 3rd party PPAs for Python 3.5 and (ii) that we'd have access to pip3.5, which would resolve the need to use the inflexible `-t` method with pip (and also would allow us to use the Ansible `pip` module instead of shelling out wityh `command`).

### CODE REVIEW

File: roles/crawler/tasks/install-python-deps.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
-- name: Install crawler.py pip dependencies to python3.5 local distribution package directory.
+- name: Install crawler.py pip dependencies.
   become: yes
-  command: pip3 install -t /usr/local/lib/python3.5/dist-packages -r crawler-requirements.txt
-  args:
-    chdir: "~{{ ssh_username }}/FingerprintSecureDrop/fpsd/requirements"
+  pip:
+    executable: pip3
```
Comment (psivesely) [inline roles/crawler/tasks/install-python-deps.yml:5]: This should be pip3.5

**Comment by conorsch** [comment]
Due to a regression in the Ubuntu Vagrant boxes, the shared folder functionality is rather broken: https://bugs.launchpad.net/cloud-images/+bug/1565985 Therefore we'll resort to a two-step provisioning run on box creation. Added an informative message so folks know what to do:

```
==> default: You must manually enable the shared folder in the Vagrantfile.
==> default: Edit the 'config.vm.synced_folder' line and set 'disabled' to 'false',
==> default: then run 'vagrant reload --provision' to reboot and mount the shared folder.
==> default: We do apologize for the inconvenience. Upstream changes are pending
==> default: to avoid this manual bootstrapping process in the future.
```

Once the upstream box is patched, we can remove the custom scripting logic. From the discussion in that thread, looks like a fix might be released in a few weeks.

File: roles/common/tasks/main.yml [hunk 2]
```diff
@@ -1,4 +1,27 @@
...
-- include: install-python35.yml
-- include: install-apt-deps.yml
+- name: Install common packages.
+  become: yes
+  apt:
+    name: "{{ item }}"
+    state: present
+    update_cache: yes
+    # Only update if it's been 24 hours since last time
+    cache_valid_time: 86400
```
Comment (garrettr) [inline roles/common/tasks/main.yml:9]: Cool, I didn't know about this before.

File: roles/crawler/tasks/install-python-deps.yml (hunk 1 above)
Comment (conorsch) [inline roles/crawler/tasks/install-python-deps.yml:5]: No, pip3 is correct:

```
ubuntu@ubuntu-xenial:~$ which pip3.5
ubuntu@ubuntu-xenial:~$ pip3 --version
pip 8.1.1 from /usr/lib/python3/dist-packages (python 3.5)
```

**Comment by conorsch** [comment]
Not quite there yet—we're still missing some pip dependencies:

```
ubuntu@ubuntu-xenial:~$ cd FingerprintSecureDrop/fpsd
ubuntu@ubuntu-xenial:~/FingerprintSecureDrop/fpsd$ ./sorter.py 
ubuntu@ubuntu-xenial:~/FingerprintSecureDrop/fpsd$ ./crawler.py 
Traceback (most recent call last):
  File "./crawler.py", line 28, in <module>
    from tbselenium.tbdriver import TorBrowserDriver
ImportError: No module named 'tbselenium'
```

**Pull request merged**